src/rez/tests/test_packages.py

```diff
@@ -309,11 +316,17 @@ def preprocess(this, data):
 
     def test_6(self):
         """test variant iteration."""
+        # ensure that differing case doesn't get interpreted as different repos
+        # on case-insensitive platforms (eg windows)
+        base = os.path.join(self.py_packages_path, "variants_py", "2.0")
+        if not platform_.has_case_sensitive_filesystem:
+            base = base.lower()
```

Comment:
should move `_format_platformpath` and reuse here instead

Reply:
Actually, this sort of code already exists in another place, see https://github.com/nerdvegas/rez/blob/master/src/rezplugins/package_repository/filesystem.py#L473.

It would be good to add a utility function to rez.utils.filesystem, something like so:

```
def canonical_path(path, platform=None):
    if platform is None:
        from rez.utils.platform_ import platform_
        platform = platform_

    path = os.path.realpath(path)
    if not platform.has_case_sensitive_filesystem:
        path = path.lower()
    return path
```

I think this may also fix a potential bug currently, where two package repositories could be the same, but if one uses symlinks to the same path on disk, rez will think they are two different repos.

To give some context: The lower() is there to fix a previous bug that was caused by differing cases of the same path, on windows (non-case-sensitive), which then led rez to believe that there were two different packages that were in fact the same package ('foo' vs 'Foo' for eg).

At least if there's just the one utility function, this can be documented in one place.

Reply:
Thank you, I agree with this idea.

I wrote `canonical_path` in addition to tests for it.

There are some *frustrating* pain points here for testing however.

``` python
# python-3.6.5, windows-10
ntpath.realpath('/a/b/c')      #>> 'C:\\a\\b\\c'
posixpath.realpath('C:/a/b/c') #>> 'C:\\Users\\willjp\\progs_shared\\vendor\\rez/C:/a/b/c'
```

``` python
# python-3.7.4, linux
ntpath.realpath('/a/b/c')     #>> '\\a\\b\\c'
posixpath.realpath('/a/b/c')  #>> '/a/b/c'
```

This change in python is not represented in the docs, so I am uncertain when they were introduced:  https://docs.python.org/3/library/os.path.html?highlight=realpath#os.path.realpath